pkg/apis/sensor/v1alpha1/types.go

```diff
@@ -358,6 +360,39 @@ type URLArtifact struct {
 	VerifyCert bool   `json:"verifyCert,omitempty" protobuf:"bytes,2,opt,name=verifyCert"`
 }
 
+// GitArtifact contains information about an artifact stored in git
+type GitArtifact struct {
+	// Git URL
+	URL string `json:"url" protobuf:"bytes,1,name=url"`
+	// Directory to clone the repository. We clone complete directory because GitArtifact is not limited to any specific Git service providers.
+	// Hence we don't use any specific git provider client.
+	CloneDirectory string `json:"cloneDirectory" protobuf:"bytes,2,name=cloneDirectory"`
+	// Creds contain reference to git username and password
+	// +optional
+	Creds *GitCreds `json:"creds,omitempty" protobuf:"bytes,3,opt,name=creds"`
```

Comment:
I've against having the credentials stored in the sensor manifest objects for security purposes. I'd rather use a reference to a secret here.

Reply:
ahh, nevermind I see that the `GitCred` is actually `corev1.SecretKeySelector`. all good